packages/grafana-ui/src/components/Pagination/Pagination.test.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Pagination } from './Pagination';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+const getPageButton = async (page: number) => {
+  return await screen.queryByText(`${page}`);
+};
+const getNextButton = async () => {
+  return await screen.getByLabelText('next');
+};
+const getPreviousButton = async () => {
+  return await screen.getByLabelText('previous');
+};
+const getAllPaginationElements = async () => {
+  return await screen.findAllByRole('listitem');
+};
+
+describe('Pagination', () => {
+  let currentPage = 1;
+  const onNavigate = (page: number) => {
+    currentPage = page;
+  };
+  const MAX_ELEMENTS = 11;
+  it('renders the correct number of pages', async () => {
+    render(<Pagination numberOfPages={3} currentPage={currentPage} onNavigate={onNavigate} />);
+    const listItems = await getAllPaginationElements();
+    expect(listItems.length).toEqual(5);
+  });
+
+  it('renders the condensed version for a large number of pages', async () => {
+    render(<Pagination numberOfPages={100} currentPage={currentPage} onNavigate={onNavigate} />);
+    const listItems = await getAllPaginationElements();
+    expect(listItems.length).toEqual(MAX_ELEMENTS);
+  });
+
+  it('increments the page', async () => {
+    currentPage = 9;
+    render(<Pagination numberOfPages={13} currentPage={currentPage} onNavigate={onNavigate} />);
+    const nextButton = await getNextButton();
```

Comment:
```suggestion
    const nextButton = getNextButton();
```
`get*` queries don't return a promise.